src/main/java/com/softserve/teamproject/entity/Student.java

```diff
@@ -111,7 +111,7 @@ public boolean equals(Object otherObject) {
 
   @Override
   public int hashCode() {
-    return Objects.hash(firstName, lastName, group.getId());
+    return Objects.hash(firstName, lastName);
```

Comment:
yes, in order to solve this problem we have to have some business key, for example 'studentNickname' which would be unique. but using other related entities in hashcode() doesn't solve the problem at all.